app/src/main/java/com/infomaniak/mail/ui/main/menu/MoveFragment.kt

```diff
@@ -17,55 +17,59 @@
  */
 package com.infomaniak.mail.ui.main.menu
 
+import android.graphics.drawable.InsetDrawable
 import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
-import androidx.core.view.isGone
-import androidx.core.view.isVisible
 import androidx.core.widget.doOnTextChanged
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.activityViewModels
 import androidx.fragment.app.viewModels
 import androidx.navigation.fragment.findNavController
 import androidx.navigation.fragment.navArgs
-import androidx.recyclerview.widget.RecyclerView
 import com.infomaniak.lib.core.utils.hideKeyboard
 import com.infomaniak.lib.core.utils.safeBinding
+import com.infomaniak.lib.core.views.DividerItemDecorator
 import com.infomaniak.mail.MatomoMail.SEARCH_DELETE_NAME
 import com.infomaniak.mail.MatomoMail.SEARCH_VALIDATE_NAME
 import com.infomaniak.mail.MatomoMail.trackCreateFolderEvent
 import com.infomaniak.mail.MatomoMail.trackMoveSearchEvent
 import com.infomaniak.mail.R
+import com.infomaniak.mail.data.cache.mailboxContent.FolderController
 import com.infomaniak.mail.data.models.Folder
-import com.infomaniak.mail.data.models.Folder.FolderRole
 import com.infomaniak.mail.databinding.FragmentMoveBinding
+import com.infomaniak.mail.ui.MainViewModel
+import com.infomaniak.mail.ui.alertDialogs.InputAlertDialog
+import com.infomaniak.mail.utils.UiUtils
+import com.infomaniak.mail.utils.extensions.bindAlertToViewLifecycle
 import com.infomaniak.mail.utils.extensions.handleEditorSearchAction
 import com.infomaniak.mail.utils.extensions.setOnClearTextClickListener
 import com.infomaniak.mail.utils.extensions.setSystemBarsColors
 import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
 
 @AndroidEntryPoint
-class MoveFragment : MenuFoldersFragment() {
+class MoveFragment : Fragment() {
 
     private var binding: FragmentMoveBinding by safeBinding()
     private val navigationArgs: MoveFragmentArgs by navArgs()
     private val moveViewModel: MoveViewModel by viewModels()
+    private val mainViewModel: MainViewModel by activityViewModels()
 
     @Inject
-    lateinit var searchFolderAdapter: FolderAdapter
+    lateinit var inputDialog: InputAlertDialog
 
-    private val searchResultsAdapter by lazy {
-        searchFolderAdapter(isInMenuDrawer, shouldIndent = false, onFolderClicked = ::onFolderSelected)
-    }
-
-    private var hasAlreadyTrackedFolderSearch = false
+    @Inject
+    lateinit var folderController: FolderController
 
-    private var currentFolderId: String? = null
+    @Inject
+    lateinit var injectedFolderAdapter: FolderAdapter
 
-    override val defaultFoldersList: RecyclerView by lazy { binding.defaultFoldersList }
-    override val customFoldersList: RecyclerView by lazy { binding.customFoldersList }
+    private val folderAdapter inline get() = binding.foldersRecyclerView.adapter as FolderAdapter
```

Comment:
We already have a reference on the adapter with the injected version. I don't think we need to create an alternative getter for it. This would also spare use from having to call one of them "injected"